### ISSUE TITLE
[failing test] test_foreach::test_binary_op_scalarlist_fastpath

### ISSUE THREAD

**Issue by kshitij12345** [pytorch/pytorch#77946]
### 🐛 Describe the bug

Following tests are failing in PR https://github.com/pytorch/pytorch/pull/77524
* `test_binary_op_scalarlist_fastpath__foreach_div_cuda_float16`
* `test_binary_op_scalarlist_fastpath__foreach_mul_cuda_float16`

CI Ref:

* https://github.com/pytorch/pytorch/runs/6500788273?check_suite_focus=true#step:9:8642
* https://github.com/pytorch/pytorch/runs/6512197886?check_suite_focus=true#step:9:9722

### Versions

master

**Comment by crcrpar** [comment]
What GPU do you use locally?
I remember I failed to reproduce and I thought the GPU architecture matter.

**Comment by kshitij12345** [comment]
The GPU on local system is `NVIDIA GeForce RTX 2060` which has arch SM75.

**Comment by ngimel** [comment]
Should we switch to checking the number of particular ops in the profile instead of checking the number of launched kernels? I don't remember if slow path launches `at::foo` ops (in which case this should work) or `at::native::foo` ops (and then it wouldn't, because profiler won't see those ops).

**Comment by crcrpar** [comment]
`t1.div(t2)` and `t1.mul(t2)` would be what slow paths use. I'm not sure which ops are called off the top of my head.

**Comment by ngimel** [comment]
Yeah that would show up in the profiler as ops, so you'll have a lot of `divs` and `muls`
